src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMutation.cs

```diff
@@ -229,6 +232,111 @@ public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[
         }
     }
 
+    internal class CollectImpl : CollectFunction, IAsyncTexlFunction3
+    {
+        public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            return await new CollectProcess().Process(irContext, args, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    internal class CollectScalarImpl : CollectScalarFunction, IAsyncTexlFunction3
+    {
+        public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            return await new CollectProcess().Process(irContext, args, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    internal class CollectProcess
+    {
+        internal async Task<FormulaValue> Process(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
```

Comment:
Should this be static? Doesn't seem like you need to construct the instance of collectprocess.